catavatar/catavatar.php

```diff
@@ -0,0 +1,306 @@
+		'$uncache' => time(),
+		'$uid' => local_user(),
+		'$usecat' => L10n::t('Use Cat as Avatar'),
+		'$morecat' => L10n::t('More Random Cat!'),
+		'$emailcat' => L10n::t('Reset to email Cat'),
+		'$seed' => PConfig::get(local_user(), "catavatar", "seed", false),
+		'$header' => L10n::t('Cat Avatar').' '.L10n::t('Settings'),
+	]);
+	return;
+}
+
+function catavatar_addon_settings_post(&$a, &$s) {
+	if(! local_user())
+		return;
+		
+	// delete the current cached cat avatar
+	$user = dba::selectFirst('user', ['email'],
+		[
+			'uid' => $uid,
+			'blocked' => 0,
+			'account_expired' => 0,
+			'account_removed' => 0,
+		]
+	);
+	$seed = PConfig::get(local_user(), "catavatar", "seed", md5(trim(strtolower($user['email']))));
+	$imageurl = preg_replace('/[^A-Za-z0-9\._-]/', '', $seed); 
+	$imageurl = substr($imageurl,0,35).'';
+	$cachefile = get_cachefile($imageurl);
+	if ($cachefile != "" && file_exists($cachefile)) {
+		unlink($cachefile);
+	}
+		
+		
+	if (x($_POST,"catavatar-usecat")) {
+		$url = $a->get_baseurl()."/catavatar/".local_user();
+		
+		// set the catavatar url as avatar url in contact and default profile
+		// and set profile to 0 to current photo
+		// I'm not sure it's the correct way to do this...
+		$r = dba::update('contact', 
+			['photo'=>$url."/4", 'thumb'=>$url."/5", 'micro'=>$url."/6", 'avatar-date'=>DateTimeFormat::utcNow()], 
+			['uid'=>local_user(), 'self'=>1]
+		);
+		if ($r===false) {
+			notice(L10n::t('There was an error, the cat ran away.'));
+			return;
+		}
+
+		$r = dba::update('profile', 
+			['photo'=>$url."/4", 'thumb'=>$url."/5"], 
+			['uid'=>local_user(), 'is-default'=>1]
+		);
+		if ($r===false) {
+			notice(L10n::t('There was an error, the cat ran away.'));
+			return;
+		}
+
+		$r = dba::update('photo', 
+			['profile'=>0], 
+			['uid'=>local_user(), 'profile'=>1]
+		);
+		if ($r===false) {
+			notice(L10n::t('There was an error, the cat ran away.'));
+			return;
+		}
+
+
+		// Update global directory in background
+		$url = $a->get_baseurl() . '/profile/' . $a->user['nickname'];
+		if ($url && strlen(Config::get('system','directory'))) {
+			Worker::add(PRIORITY_LOW, "Directory", $url);
+		}
+
+		Worker::add(PRIORITY_LOW, 'ProfileUpdate', local_user());
+		
+		info(L10n::t("Meow!"));
+		return;
+	}
+	
+
+
+	if (x($_POST,"catavatar-morecat")) {
+		PConfig::set(local_user(), "catavatar", "seed", time());
+	}
+
+	if (x($_POST,"catavatar-emailcat")) {
+		PConfig::delete(local_user(), "catavatar", "seed");
+	}
+}
+
+
+/**
+ * Returns the URL to the cat avatar
+ *
+ * @param $a array
+ * @param &$b array
+ */
+function catavatar_lookup($a, &$b) {
+	$user = dba::selectFirst('user', ['uid'],['email'=>$b['email']]);
+	
+	$url = $a->get_baseurl().'/catavatar/'.$user['uid'];
+
+	switch($b['size']) {
+		case 175: $url.="/4"; break;
+		case 80: $url.="/5"; break;
+		case 47: $url.="/6"; break;
+	}
+
+	$b['url'] = $url;
+	$b['success'] = true;
+}
+
+
+function catavatar_module(){}
```

Comment:
Standards: Please add a space before opening braces.

Reply:
as is a empty function, what's the standard here?

```php
function catavatar_module()
{}

// or

function catavatar_module()
{
}
```

Reply:
I would go with `function catavatar_module() {}` as it just is a (wrong) way of indicating the addon provides a module.

Reply:
Yeah. Ok.
( in a future hopefully we will see something like
```php
class CatAvatar extends Addon implements Module {
```
:-D )

Reply:
I'd rather see
```php
class CatAvatar extends Addon
{
    public function __construct(\Friendica\App $a)
    {
        $a->addRoute('catavatar', '\Friendica\Addon\Module\CatAvatar');
        ...
    }
    ...
}
```
and have the module in a separate file in the addon.